django/contrib/sessions/backends/base.py

```diff
@@ -87,33 +88,14 @@ def test_cookie_worked(self):
     def delete_test_cookie(self):
         del self[self.TEST_COOKIE_NAME]
 
-    def _hash(self, value):
-        key_salt = "django.contrib.sessions" + self.__class__.__name__
-        return salted_hmac(key_salt, value).hexdigest()
+    def _salt(self):
+        return "django.contrib.sessions" + self.__class__.__name__
 
-    def encode(self, session_dict):
-        "Return the given session dictionary serialized and encoded as a string."
-        serialized = self.serializer().dumps(session_dict)
-        hash = self._hash(serialized)
-        return base64.b64encode(hash.encode() + b":" + serialized).decode('ascii')
+    def encode(self, session_dict, compress=False):
```

Comment:
Change of function signature needs documenting as it will affect third-party session backends.

Reply:
Perhaps additional arguments can be provided as `**extra` which will accept anything supported by `signing.dumps()`?